clippy2021.md

```diff
@@ -0,0 +1,178 @@
+nightly Rust nowadays makes it more more probably that a lint with many FPs hits
+stable. This leads to annoyed users, that will disable these new lints in the
+best case and to more annoyed users, that will stop using Clippy in the worst.
+We should think of a process to prevent this from happening. This will also be
+addressed later in this issue.
+
+[#6429](https://github.com/rust-lang/rust-clippy/issues/6429)
+
+## Internal
+
+(The end of) 2020 has shown, that Clippy has to think about the available
+resources, especially regarding management and maintenance of the project. This
+section address issues affecting team members and contributors.
+
+### Management
+
+In 2020 we got over 1000 open issues with now between 25-35 open PRs. This is
+mostly because of the lack of resources in the team. According to the planned
+roadmap of the Rust project, we should improve our processes and look for more
+eyes and hands to help reviewing and triaging PRs and issues.
+
+#### Clear expectations for team members
+
+We will have to write this out according to the planned Rust 2021 roadmap
+anyway. This should not put more pressure on the team members, but rather help
+them and interested folks to know what our expectations are.
+
+#### More maintainer
+
+More people means less work for each individual. We should work out a process
+how onboarding of new folks can work and look out for people that are interested
+joining and taking up responsibility.
+
+#### (Bi-)weekly meetings
+
+Other teams have regular meetings. Clippy is big enough, that it might be worth
+it to also do them. Especially if we want/get more people joining the team, this
+can be important for sync-ups. I for example can't keep track of all the new
+lints and features that are implemented in Clippy, because I stopped looking at
+every PR a while ago. And no one should have to do this to be up-to-date.
+
+### Development
+
+We can always improve the developer and contributor experience with Clippy and
+simultaneously make the life of the team members easier.
+
+#### Process for new lints
+
+As already mentioned above, classifying new lints gets quite hard, because the
+probability of a buggy lint getting into stable is quite high. We should work
+out a process (and maybe tooling) to make sure we test lints, before releasing
+them into the wild.
+
+[#6429 (comment)](https://github.com/rust-lang/rust-clippy/issues/6429#issuecomment-741056379)
+
+#### RFC/MCP/FCP process
+
+Maybe related to the point before, we should consider implementing some process
+for major changes. This would also be helpful for this list, since it currently
+is a semi-organized dump of ideas. Some points raised here may require some
+process, before implementing them.
+
+#### Dev-Tools
+
+We already have `cargo dev` which makes Clippy development easier. We should
+expand its capabilities.
+
+[#5394](https://github.com/rust-lang/rust-clippy/issues/5394)
+
+#### `rustc` integration
```

Comment:
Some general issues I think we should address, that would involve some communication with the compiler team:

* Implement a solution for the problem of conflicting lints, probably directly in rustc?
* Clarify the situation with pre-expansion lints. This comes up quite often.

Reply:
Another thing that came to mind: Implementing the `#[expect(<lint>)]` attribute, which warns if the lint you expect to trigger doesn't trigger (anymore). That can help, if people allow lints because of FPs, but then never remove the ``allow`` even when the lint gets fixed.

Reply:
But I think this is also rather answering the "How" question, than the "What".